src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -114,6 +115,50 @@ def new_forward(*inputs):
 
         layer.module.forward = new_forward
 
+    def export_model(self, model, save_path=None, onnx_path=None, input_shape=None):
+        """
+        Export pruned model weights, masks and onnx model(optional)
+
+        Parameters
+        ----------
+        model : pytorch model
+            pytorch model to save
+        save_path : str
+            path to save state_dict and mask_dict
+        onnx_path : str
+            (optional) path to save onnx model
+        input_shape : list or tuple
+            input shape to onnx model
+        """
+        export_dict = {}
+        export_dict['state_dict'] = model.state_dict()
+        mask_dict = {}
+        for layer_name in self.mask_list:
+            mask_dict[layer_name + '.weight'] = self.mask_list[layer_name]
+        export_dict['mask_dict'] = mask_dict
+        for name, m in model.named_modules():
+            mask = self.mask_list.get(name)
+            if mask is not None:
+                mask_sum = mask.sum().item()
+                mask_num = mask.numel()
+                _logger.info('Layer: {}  Sparsity: {}'.format(name, 1 - mask_sum / mask_num))
+                print('Layer: {}  Sparsity: {}'.format(name, 1 - mask_sum / mask_num))
+                m.weight.data = m.weight.data.mul(mask)
+            else:
+                _logger.info('Layer: {}  Sparsity: {}'.format(name, 0))
+                print('Layer: {}  Sparsity: {}'.format(name, 0))
+        assert save_path is not None, 'Save Path must be specified'
```

Comment:
better move this input checking assertion to the begining of the function.

Reply:
@chicm-ms I've fixed this.